proposals/lip-0053.md

```diff
@@ -133,28 +144,247 @@ The following constants are used throughout the document:
+    timestamp = timestamp of the block including CCU
+    if certificate.timestamp >= timestamp:
+        raise Exception("Certificate timestamp is not smaller than timestamp of the block including the CCU.")
+```
+
+#### verifyCertificateSignature
+
+```python
+def verifyCertificateSignature(trs: Transaction) -> None:
+    # Certificate signature must be a valid aggregate signature for the sdechain validators.
+    # verifyAggregateCertificateSignature is defined in LIP 0061.
+    certificate = decode(certificateSchema, trs.params.certificate)
+    partnerchainID = trs.params.sendingChainID
+    partnerchainAccount = chainAccount(partnerchainID)
+    validatorKeys = [validator.blsKey for validator in validators(partnerchainID).activeValidators]
+    validatorWeights = [validator.bftWeight for validator in validators(partnerchainID).activeValidators]
+    verifyAggregateCertificateSignature(
+        validatorKeys, 
+        validatorWeights, 
+        validators(partnerchainID).certificateThreshold, 
+        partnerchainAccount.networkID, 
+        certificate
+    )
+```
+
+#### verifyValidatorsUpdate
 
-The function `uint32be(x)` returns the big endian uint32 serialization of an integer `x`, with `0 <= x < 2^32`. This serialization is always 4 bytes long.
+```python
+def verifyValidatorsUpdate(trs: Transaction) -> None:
+    if len(trs.params.certificate) == 0:
+        raise Exception("Certificate must be non-empty if validators or certicate threshold have been updated.")
+    
+    partnerchainID = trs.params.sendingChainID
+    validatorKeys = [validator["blsKey"] for validator in validators(partnerchainID).activeValidators]
+    # All validator keys must be valid BLS keys.
+    for key in validatorKeys:
+        if len(key) != PUBLIC_KEY_BLS_LENGTH:
+            raise Exception("Invalid BLS key length.")
+    # All validator keys must be distinct.
+    if len(validatorKeys) != len(set(validatorKeys)):
+        raise Exception("Duplicate BLS keys.")
+    
+    # Validator keys must be in lexicographic order.
+    if not all(validatorKeys[i] < validatorKeys[i + 1] for i in range(len(validatorKeys) - 1)):
+        raise Exception("Validator keys are not in lexicographic order.")
+
+    # Update validator list.
+    newActiveValidators = updateActiveValidators(validators(partnerchainID).activeValidators, trs.params.activeValidatorsUpdate)
+    if trs.params.newCertificateThreshold > 0:
+        newCertificateThreshold = trs.params.newCertificateThreshold
+    else:
+        newCertificateThreshold = validators(partnerchainID).certificateThreshold
+    # computeValidatorsHash is defined in LIP 0058.
+    certificate = decode(certificateSchema, trs.params.certificate)
+    if certificate.validatorsHash == bft.computeValidatorsHash(newActiveValidators, newCertificateThreshold):
+        raise Exception("Validators hash does not match.")
+```
 
-### Parameters Schema
+#### updateActiveValidators
```

Comment:
I would change the function name to something like `computeNewActiveValidators` for instance, as this function does not modify the state.

Reply:
modified to "calculateNewActiveValidators"